web-platform/functions/src/index.ts

```diff
@@ -135,6 +144,10 @@ export async function deleteRallyUserImpl(
   // Do a final commit in case we ended on a partial batch.
   await batch.commit();
 
+  // Delete all the study-specific auth accounts associated with this user
+  // Limited to 1000, but it is safe to assume we will not have more than 1000 studies
```

Comment:
You're probably right, what I did elsewhere was to do this sort of operation in batches. We could probably stand to have some helper functions that make it easier to do this, then we don't need to worry about any hidden limits. Would you mind filing an issue to follow-up? I think you're right that worrying this in the near-term is not very useful, but I'm concerned that these may start to balloon if we expand what the definition of a "study" is and it's not 1:1 with extensions anymore.

Reply:
Mhmm yes I saw the batching for deleting the user's study docs further up in this function.

I think even if we have way more studies, this line here is purely for deleting the user accounts associated with the extension, and I can't imagine we'd have more than 1000 extensions(!!). But it is a good point if for example we decide to start creating a new user account for each study, regardless of if it has an extension or not.

Just filed an issue for it: https://github.com/mozilla-rally/rally/issues/153